### ISSUE TITLE
Compose error when using env_file and environment uses object syntax in yml

### ISSUE THREAD

**Issue by robpc** [aws/amazon-ecs-cli#124]
Given a project like:

_docker-compose.yml_

```
version: '2'
services:
  test:
    image: busybox
    environment:
      B: d
    env_file:
      - config.env
```

_config.env_

```
A=1
```

`ecs-cli compose up` returns the stacktrace

```
22:30 # ecs-cli compose up
panic: interface conversion: interface is map[interface {}]interface {}, not []interface {}

goroutine 1 [running]:
panic(0x/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.c/usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/config/merge.go:100 +0x42c0165120, 0x12, 0xc82015fa10, 0x8, 0x0, 0x0)
github.c/usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/config/merge_v2.go:95 +0x83b0d0, 0xc820165540, 0xc820165120, 0x12, 0xc82015fa10, 0xc82015f9b0, 0x10e36e0, 0xc
github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/config.MergeServicesV2(0xc820165580, 0x7f994b3cb0d0, 0xc820165540, 0x7f994b3cb0f8, 0x110a198, 0xc820165120, 0x12, 0xc82015d400, 0x6f,
0x26f, ./usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/config/merge_v2.go:36 +0x2e3
github.c/usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/config/merge.go:40 +0x24c65540, 0x7f994b3cb0f8, 0x110a198, 0xc820165120, 0x12, 0xc82015d400, 0x6f, 0x26f, ...
github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/project.(*Project).load(0xc820163a00, 0xc820165120, 0x12, 0xc82015d400, 0x6f, 0x26f, 0x0, 0x0)
        /usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/docker/libcompose/project/project.go:84 +0x323
github.com/aws/amazon-ecs-cli/ecs-cli/modules/compose/ecs.(*ecsProject).parseCompose(0xc820163a00, 0x0, 0x0)a00, 0x0, 0x0)
        /usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/modules/compose/ecs/project.go:105 +0xde
github.com/aws/amazon-ecs-cli/ecs-cli/modules/compose/cli/ecs/app.projectFactory.loadProject(0x7f994b3cb210, 0xc820163a00, 0x0, 0x0)
        /usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/modules/compose/cli/ecs/app/factory.go:50 +0xe63
github.com/aws/amazon-ecs-cli/ecs-cli/modules/compose/cli/ecs/app.(*projectFactory).Create(0x110a198, 0xc8201741e0, 0xc820141e00, 0x0, 0x0, 0x0, 0x0)
        /usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/modules/compose/cli/ecs/app/app.go:45 +0x6a
github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/codegangsta/cli.Command.Run(0xc37008, 0x2, 0x0, 0x0, 0x0, 0x0, 0x0, 0xdcd020, 0xac, 0x0, ...)
        /usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/codegangsta/cli/app.go:257 +0xdacx1055
github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/codegangsta/cli.Command.startApp(0xc341d8, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xd52ac0, 0x39, 0x0, ...)
        /usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/codegangsta/cli/command.go:55 +0x15f3
github.com/aws/amazon-ecs-cli/ecs-cli/vendor/github.com/codegangsta/cli.(*App).Run(0xc8200b6b00, 0xc82000a120, 0x3, 0x3, 0x0, 0x0)c0, 0x39, 0x0, ...)
        /usr/src/app/src/github.com/aws/amazon-ecs-cli/ecs-cli/main.go:45 +0x9fa/codegangsta/cli/app.go:159 +0xc34
```

Switching the `docker-compose.yml` file to use array syntax for the `environment` block removes this error.

```
version: '2'
services:
  test:
    image: busybox
    environment:
      - B=d
    env_file:
      - config.env
```

**Comment by uttarasridhar** [comment]
Thank you for the bug report. This is currently being addressed in our dependency : https://github.com/docker/libcompose/pull/307.
In a interim, please continue using environment format key=value. Apologies for the inconvenience.

**Comment by pgeez** [comment]
@uttarasridhar Any idea on when this will be fixed?

+1 for this issue.

**Comment by uttarasridhar** [comment]
Thanks for reporting the bug. This issue was fixed in the latest pull request #158.
